src/main/java/com/icthh/xm/ms/entity/service/impl/XmEntityServiceImpl.java

```diff
@@ -211,10 +211,11 @@ public XmEntity saveXmEntity(XmEntity xmEntity) {
     }
 
     private void processName(XmEntity xmEntity) {
-        TypeSpec typeByKey = xmEntitySpecService.findTypeByKey(xmEntity.getTypeKey());
-        if (StringUtils.isNotEmpty(typeByKey.getNamePattern())) {
-            xmEntity.setName(simpleTemplateProcessors.processTemplate(typeByKey.getNamePattern(), xmEntity));
-        }
+        xmEntitySpecService
```

Comment:
please consider this approach (avoids complex lambda in ifPresent): 
```
Optional.ofNullable(xmEntity.getTypeKey())
                .flatMap(xmEntitySpecService::getTypeSpecByKey)
                .map(TypeSpec::getNamePattern)
                .filter(StringUtils::isNotEmpty)
                .map(parentName -> simpleTemplateProcessors.processTemplate(parentName, xmEntity))
                .ifPresent(xmEntity::setName);
```

Reply:
fixed